src/vs/vscode.d.ts

```diff
@@ -5399,6 +5399,23 @@ declare module 'vscode' {
 		export function registerTaskProvider(type: string, provider: TaskProvider): Disposable;
 	}
 
+	export interface IConfigurationChangeEvent {
+
+		/**
+		 * Affected configuration keys
+		 */
+		affectedKeys: string[];
```

Comment:
`readonly`

Reply:
Maybe call this 'changedKeys' or make this an object/map like `changes: Map<string, any>`. The likelihood of someone calling `getValue` for the changed keys is high IMO

Reply:
If we provide the map/object, then users might try asking values for `sections` because we highly use it in other places which can bloat up the event

I think having one API for one thing makes it simple and easy to use.  For eg., you can always call a function that reads configuration initially and on configuration change.

Reply:
Calling `changedKeys` might mislead that keys got changed right? Instead, it's the values for the keys got changed.